### ISSUE TITLE
[Android][A11y] Custom scroll View a11y focus jumps randomly

### ISSUE THREAD

**Issue by chunhtai** [flutter/flutter#87305]
b/194974902

## Steps to Reproduce

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter accessibility issues demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
        visualDensity: VisualDensity.adaptivePlatformDensity,
      ),
      home: MyHomePage(title: 'Flutter accessibility issues demo page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);
  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("Demo"),
      ),
      body: CustomScrollView(
        slivers: [
          SliverToBoxAdapter(child: SingleChildScrollView(scrollDirection: Axis.horizontal,child: IntrinsicWidth(child:Row(children: [
            Container(
              width:300,
                height:300,
                child: Text('box'),),
            Container(
              width:300,
              height:300,
              child: Text('box'),),
            Container(
              width:300,
              height:300,
              child: Text('box'),),
            Container(
              width:300,
              height:300,
              child: Text('box last one'),)
          ],)))), /// Scrollable horizontally
          SliverList(
            delegate: SliverChildListDelegate(
              [
                Container(
                  width:300,
                  height:200,
                  child: Text('first box'),),
                Container(
                  width:300,
                  height:200,
                  child: Text('box'),),
                Container(
                  width:300,
                  height:200,
                  child: Text('box'),),
                Container(
                  width:300,
                  height:200,
                  child: Text('box'),),
                Container(
                  width:300,
                  height:200,
                  child: Text('box'),),
              ]
            ),
          )                                        /// Scrollable vertically
        ],
      ),
    );
  }
}

```
1. turn on talkback
2. swipe to focus the `box last one` in the first horizontal scroll view
3. swipe right to focus the next box

expect: `first box` should be focused
actual: the focus jumped to a random box in the second scroll view

## Logs

<details>
<summary>Logs</summary>

```
```

```
```

</details>

**Comment by chunhtai** [comment]
This is a bug in talkback, I can reproduce it in native android app as well. https://github.com/chunhtai/scrollable-bug

**Comment by chunhtai** [comment]
Filed b/195134536
